src/WANN.cpp

```diff
@@ -123,6 +127,49 @@ class WANN {
     return z ;
   }
   
+  List queryFRANN(const ANNpointArray query, const int nq, const int k, const double rad, const double eps=0.0) {
+
+    // build tree (in case we didn't already)
+    build_tree();
+
+    // ANN style arrays to hold return values for one point
+    ANNidxArray nn_idx = new ANNidx[k];
+    ANNdistArray dists = new ANNdist[k];
+
+    // declare matrices for return values here
+    NumericMatrix rdists(nq, k);
+    IntegerMatrix ridx(nq, k);
```

Comment:
Can you think of a way to avoid declaring these if k==0? Does it matter?

Reply:
Okay that sounds great.

We could have an optional argument which is the fixed radius.
When radius is set, the list will contain the extra vector containing counts of points within the fixed radius.

We could have a simplified path through the code in case k==0 which would not make and returning the dists and idx matrices. That could in theory break someone's code... but if they put k==0, they get absolutely useless result anyway, so it's probably fine to make that change.

If the user sets both the fixed radius and K, the ANN library still supports that, and they simply get both outputs filled in.